indico/modules/rb/client/js/modules/admin/MapAreasPage.jsx

```diff
@@ -0,0 +1,372 @@
+                            )}
+                        </>
+                    </Tooltip>
+                </Rectangle>
+            );
+        });
+    };
+
+    onSubmit = (data) => {
+        const {actions: {createMapArea, updateMapAreas}} = this.props;
+        const {areaProps: {id}} = this.state;
+        let result;
+
+        if (!id) {
+            result = createMapArea(data);
+        } else {
+            result = updateMapAreas([{...data, id}]);
+        }
+
+        result.then(() => {
+            this.setState({areaProps: null});
+        });
+    };
+
+    renderModalForm = () => {
+        const {areaProps} = this.state;
+        if (!areaProps) {
+            return null;
+        }
+
+        const {id, bounds, name, isDefault} = areaProps;
+        return (
+            <FinalForm onSubmit={this.onSubmit}
+                       render={this.renderForm}
+                       initialValues={{
+                           name: !id ? null : name,
+                           bounds,
+                           default: isDefault
+                       }} />
+        );
+    };
+
+    renderForm = (fprops) => {
+        const {
+            submitting, submitSucceeded, pristine, hasValidationErrors, handleSubmit
+        } = fprops;
+        const {areaProps: {id}} = this.state;
+        const isNewArea = !id;
+
+        return (
+            <Modal onClose={() => this.setState({areaProps: null})}
+                   closeIcon
+                   open>
+                <Modal.Header>
+                    {isNewArea ? (
+                        <Translate>Create new area</Translate>
+                    ) : (
+                        <Translate>Update area</Translate>
+                    )}
+                </Modal.Header>
+                <Modal.Content>
+                    <Form id="map-area-form" onSubmit={handleSubmit}>
+                        <Form.Group>
+                            <FinalInput name="name"
+                                        label={Translate.string('Name of the area')}
+                                        disabled={submitting || submitSucceeded}
+                                        autoFocus
+                                        required />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalCheckbox name="default"
+                                           label={Translate.string('Default map area')}
+                                           toggle />
+                        </Form.Group>
+                        <Form.Group>
+                            <FinalInput name="bounds.southWest.lat"
+                                        type="number"
+                                        label={Translate.string('SW Latitude')}
+                                        disabled={submitting || submitSucceeded}
+                                        validate={v.number()}
```

Comment:
This shouldn't be needed; `input type=number` never returns non-numeric values so this validator will never fail.

Reply:
Yeah, but in case someone tampers with the `type` attribute, it will still be possible to send it to backend, causing a DB error. So, additional layer of validation doesn't hurt here.

Reply:
You know what they say about client-side validation... it's just for user convenience. If someone tampers with the attribute, they can just send random garbage as well.

The schema on the server should be more specific than just `Dict` though, to avoid this :)

Reply:
Yeah, of course it doesn't stop the user from tampering with the data even more. But isn't it better at least to have clear error messages? :P (I mean I don't see a big problem with having this extra check there)

BTW: good point regarding the schema

Reply:
I don't see the point in adding a check that will never be triggered except when the user messes around with the DOM.

Also, if we want such a validator it'd be more appropriate to add inside FinalInput so it applies to all number fields. But like I said, I don't think there's a real benefit there (I wanted to add it there initially to show a nicer error if someone enters non-numeric data, but since that doesn't work...)